modules/openapi-generator/src/test/java/org/openapitools/codegen/kotlin/AbstractKotlinCodegenTest.java

```diff
@@ -83,4 +83,19 @@ public void isDataTypeString(){
         assertTrue(codegen.isDataTypeString("kotlin.String"));
         assertTrue(codegen.isDataTypeString("String"));
     }
+
+    @Test
+    public void toModelNameShouldUseProvidedMapping() throws Exception {
+        codegen.importMapping().put("json_myclass", "com.test.MyClass");
+        assertEquals("com.test.MyClass", codegen.toModelName("json_myclass"));
+    }
+
+    @Test
+    public void convertModelName() throws Exception {
+        assertEquals(codegen.toModelName("name"), "Name");
+        assertEquals(codegen.toModelName("$name"), "Dollarname");
+        assertEquals(codegen.toModelName("nam#e"), "NamHashe");
+        assertEquals(codegen.toModelName("$another-fake?"), "DollaranotherMinusfakeQuestionMark");
```

Comment:
@karismann quick question if you don't mind. Should it be `DollarAnotherMinusFakeQuestionMark` (capital letter A and F) instead of `DollaranotherMinusfakeQuestionMark`?

Reply:
@wing328 I think that's happening because of `camelize`, after which point we would lose context of what should be capitalized here, I would think.

See https://github.com/OpenAPITools/openapi-generator/blob/b1dc2eeaacfb33f491683fab94e4d5247f8daeb4/modules/openapi-generator/src/test/java/org/openapitools/codegen/utils/StringUtilsTest.java#L26

Reply:
OK. If my memory services me well, we've something like `$prefix_$name` before going through `camelize` to handle the case mentioned above in some other generators (e.g. Perl). We can definitely go with this implementation in Kotlin and see if that's what the user expects.